transport/serialize/cborserializer.go

```diff
@@ -44,3 +44,23 @@ func (s *CBORSerializer) Deserialize(data []byte) (wamp.Message, error) {
 	}
 	return listToMsg(wamp.MessageType(typ), v)
 }
+
+// SerializeDataItem encodes any object/structure into a json payload.
+func (s *CBORSerializer) SerializeDataItem(item interface{}) ([]byte, error) {
+	var b []byte
+	return b, codec.NewEncoderBytes(&b, ch).Encode(item)
```

Comment:
hmm NewEncoderBytes can reallocate the buffer. I wonder if this is correct in case this happens. IIRC the evalution of left to right.

Reply:
@bboozzoo you mean returning?
```
var b []byte
err := codec.NewEncoderBytes(&b, ch).Encode(item)
return b, err
```
Something like that?

Reply:
yes